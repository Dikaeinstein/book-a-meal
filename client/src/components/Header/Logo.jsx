import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const Logo = ({ text, className }) => (
  <NavLink
    to="/"
    activeClassName="active"
  >
    <h1 className={className}>{text}</h1>
  </NavLink>
);

Logo.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Logo;
