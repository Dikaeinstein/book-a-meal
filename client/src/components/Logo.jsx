import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { userSignOut } from '../actions/userActions';

const ConnectedLogo = ({ text, className, goHome }) => {
  const handleClick = () => {
    goHome();
  };
  return (
    <NavLink to="/" onClick={handleClick}>
      <h1 className={className}>{text}</h1>
    </NavLink>
  );
};

ConnectedLogo.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  goHome: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  goHome: () => dispatch(userSignOut()),
});

const Logo = connect(null, mapDispatchToProps)(ConnectedLogo);

export default Logo;
