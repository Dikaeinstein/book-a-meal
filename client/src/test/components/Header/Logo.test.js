import React from 'react';
import { shallow } from 'enzyme';
import Logo from '../../../components/Header/Logo';


describe('<Logo />', () => {
  it('should match snapshot', () => {
    const tree = shallow(<Logo text="test" className="test-class" />);
    expect(tree).toMatchSnapshot();
  });
});
