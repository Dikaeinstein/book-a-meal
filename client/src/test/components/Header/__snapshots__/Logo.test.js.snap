// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should match snapshot 1`] = `
<NavLink
  activeClassName="active"
  ariaCurrent="true"
  exact={true}
  to="/"
>
  <h1
    className="test-class"
  >
    test
  </h1>
</NavLink>
`;
